zipkin-collector-service/src/main/scala/com/twitter/zipkin/collector/Main.scala

```diff
@@ -14,31 +14,35 @@
  *  limitations under the License.
  *
  */
-package com.twitter.zipkin.collector
+import com.twitter.zipkin.thriftscala.{Span => ThriftSpan}
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.server.TwitterServer
+import com.twitter.util.{Await, Future}
+import com.twitter.zipkin.cassandra.CassieSpanStoreFactory
+import com.twitter.zipkin.collector.{SpanReceiver, ZipkinQueuedCollectorFactory}
+import com.twitter.zipkin.common._
+import com.twitter.zipkin.receiver.kafka.KafkaSpanReceiverFactory
+import com.twitter.zipkin.storage.WriteSpanStore
+import com.twitter.zipkin.zookeeper.ZooKeeperClientFactory
+import kafka.serializer.Decoder
 
-import com.twitter.logging.Logger
-import com.twitter.ostrich.admin.{ServiceTracker, RuntimeEnvironment}
-import com.twitter.util.Eval
-import com.twitter.zipkin.collector.builder.CollectorServiceBuilder
-import com.twitter.zipkin.BuildProperties
+import com.twitter.zipkin.receiver.kafka.SpanDecoder
 
+object ZipkinKafkaCollectorServer extends TwitterServer
+  with ZipkinQueuedCollectorFactory
+  with CassieSpanStoreFactory
+  with ZooKeeperClientFactory
+  with KafkaSpanReceiverFactory
+{
+  def newReceiver(receive: Seq[ThriftSpan] => Future[Unit], stats: StatsReceiver): SpanReceiver =
+    newKafkaSpanReceiver(receive, stats.scope("kafkaSpanReceiver"), new SpanDecoder())
 
-object Main {
-  val log = Logger.get(getClass.getName)
+  def newSpanStore(stats: StatsReceiver): WriteSpanStore =
+    newCassandraStore(stats.scope("cassie"))
 
-  def main(args: Array[String]) {
-    log.info("Loading configuration")
-    val runtime = RuntimeEnvironment(BuildProperties, args)
-    val builder = (new Eval).apply[CollectorServiceBuilder[Seq[String]]](runtime.configFile)
-    try {
-      val server = builder.apply().apply(runtime)
-      server.start()
-      ServiceTracker.register(server)
-    } catch {
-      case e: Exception =>
-        e.printStackTrace()
-        log.error(e, "Unexpected exception: %s", e.getMessage)
-        System.exit(0)
-    }
+  def main() {
+    val collector = newCollector(statsReceiver)
+    onExit { collector.close() }
```

Comment:
code provided by [twitter server](https://twitter.github.io/twitter-server/) to register code to execute onexit